tools/stimulus.py

```diff
@@ -243,13 +247,13 @@ def generate(self, inputs, output):
         out.close()
 
     def generate_function(self, f, out):
-        print "Error: invalid code generator, this method should be overridden"
+        print("Error: invalid code generator, this method should be overridden")
```

Comment:
Idiomatic Python 3.x is to 1) `raise NotImplementedError` here, 2) mark the `generate_function` method with the `@abstractmethod` decorator and 3) construct the class as `class CodeGenerator(metaclass=ABCMeta)` (where `ABCMeta` is imported from `abc`). This prevents instantiation of `CodeGenerator` until the method is overridden.

Reply:
Yes, good point. I haven't gone through all the code to make it Python 3.x idiomatic, just functional. But this might be a good occasion to revise the entire code.

Reply:
I'd be happy to check for Python-ness of code here and in `python-igraph`, if you guys would like me to

Reply:
This PR only involves `stimulus.py`, which is the tool that generates the code for interfaces. At the moment it is mostly used by the R interface, but in principle we could also use it to build any other interface (e.g. also for `python-igraph`). I wouldn't go through the entire `python-igraph` library, but if you want to check this `stimulus.py`, please go ahead! Feel free to push immediately to this PR (or open a PR against my branch :smile: ).